### PR TITLE
Add warning if changing flutterEmbedding string.

### PR DESCRIPTION
**Pull request by reidbaker**
No bug

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by chunhtai — APPROVED** [review]
LGTM

**Review by gmackall — APPROVED** [review]

File: packages/flutter_tools/lib/src/project.dart [hunk 1]
```diff
@@ -861,8 +861,8 @@ $javaGradleCompatUrl
     }
     for (final XmlElement metaData in document.findAllElements('meta-data')) {
       final String? name = metaData.getAttribute('android:name');
-      // External code checks for this string to indentify flutter android apps. 
-      // See https://critique.corp.google.com/cl/667760684 as an example. 
+      // External code checks for this string to indentify flutter android apps.
+      // See https://critique.corp.google.com/cl/667760684 as an example.
```
Comment (johnmccutchan) [inline packages/flutter_tools/lib/src/project.dart:865]: can you replace this with just "cl/6677..." ? That is drop the full domain name.

Comment (reidbaker) [reply packages/flutter_tools/lib/src/project.dart]: Done

**Comment by reidbaker** [comment]
I am debating merging on red and ignoring "customer linux customer testing" because this pr adds a comment to android code by definition it can't impact customer code. 

WIll try a 4th rerun and then merge.

**Pull request merged**